packages/core/src/browser/tree/tree-decorator.ts

```diff
@@ -30,25 +62,41 @@ export interface TreeDecorator {
     readonly id: string;
 
     /**
-     * Fired when this decorator has calculated all the decoration data for the tree nodes. Keys are the unique identifier of the tree nodes.
+     * Fired when this decorator has calculated all the `decoration data` for the tree nodes.
      */
     readonly onDidChangeDecorations: Event<(tree: Tree) => Map<string, TreeDecoration.Data>>;
 
     /**
-     * Returns with the current decoration data for the tree argument.
+     * Computes the current `decoration data` for the given tree. Might return a promise if the computation is handled asynchronously.
      *
      * @param tree the tree to decorate.
+     *
+     * @returns (a promise of) a map containing the current {@linkTreeDecoration.Data} for each node. Keys are the unique identifier of the tree nodes.
      */
     decorations(tree: Tree): MaybePromise<Map<string, TreeDecoration.Data>>;
 
 }
 
+export const TreeDecoratorService = Symbol('TreeDecoratorService');
+
 /**
- * Decorator service which emits events from all known tree decorators.
- * Keys are the unique tree node IDs and the values
- * are the decoration data collected from all the decorators known by this service.
+ * The {@link TreeDecoratorService} manages a set of known {link TreeDecorator}s and emits events when
+ * any of the known decorators has changes. Typically, a `TreeDecoratorService` provides a contribution point that can be used to
+ * register {@link TreeDecorator}s exclusively for this service.
+ * e.g. the {@link NavigatorDecoratorService} provides the  {@link NavigatorTreeDecorator} contribution point to register custom decorators.
```

Comment:
I think we can exclude this comment, `NavigatorDecoratorService` does not exist in `@theia/core`.

Reply:
Done